graph/multi/weighted_undirected.go

```diff
@@ -0,0 +1,308 @@
+// Copyright ©2014 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package multi
+
+import (
+	"fmt"
+
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/internal/uid"
+)
+
+var (
+	wug *WeightedUndirectedGraph
+
+	_ graph.Graph                        = wug
+	_ graph.Undirected                   = wug
+	_ graph.WeightedUndirected           = wug
+	_ graph.Multigraph                   = wug
+	_ graph.UndirectedMultigraph         = wug
+	_ graph.WeightedUndirectedMultigraph = wug
+)
+
+// WeightedUndirectedGraph implements a generalized undirected graph.
+type WeightedUndirectedGraph struct {
+	EdgeWeightFunc func([]graph.WeightedLine) float64
+
+	nodes map[int64]graph.Node
+	lines map[int64]map[int64]map[int64]graph.WeightedLine
+
+	nodeIDs uid.Set
+	lineIDs uid.Set
+}
+
+// NewWeightedUndirectedGraph returns an WeightedUndirectedGraph with the specified self and absent
```

Comment:
Done

Yes, those default values in graphs in simple allow the `graph.Weighter` interface to be satisfied with user-defined values. That user hook here is provided through the `EdgeWeightFunc` field. I could have done the same thing in the constructor here as in simple - providing the func via the constructor, but this seemed nicer since there is only one value to pass and you don't need to be able to see two consistent values being passed at the same time. If you think I should change this, please let me know.

Reply:
Yes, this is nicer.